src/Sylius/Bundle/UserBundle/Provider/AbstractUserProvider.php

```diff
@@ -67,8 +66,11 @@ public function loadUserByUsername($username): UserInterface
      */
     public function refreshUser(UserInterface $user): UserInterface
     {
-        /** @var SyliusUserInterface $user */
-        Assert::isInstanceOf($user, SyliusUserInterface::class);
+        if(!$user instanceof SyliusUserInterface) {
```

Comment:
I'm wondering if it shouldn't be part of `supportsClass` function

Reply:
I don't believe, `supportsClass` does not have the same purpose.